### PR TITLE
fix(restify): upgrade package to fix type problem #1132

### PR DESCRIPTION
**Pull request by jscherer92**
## Which problem is this PR solving?
#1132 

## Short description of the changes
Updated the Restify package for @opentelemetry/instrumentation-restify

## Checklist

- [ x] Ran `npm run test-all-versions` for the edited package(s) on the latest commit if applicable.

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-restify/package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
     "gts": "3.1.0",
     "mocha": "7.2.0",
     "nyc": "15.1.0",
-    "restify": "4.3.4",
+    "restify": "8.6.1",
```
Comment (blumamir) [inline plugins/node/opentelemetry-instrumentation-restify/package.json:57]: Thank you for submitting a fix. I think the original error originated from [`@types/restify` here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/ec9ee131635dc2db88deea4f2efb887ff6f60577/plugins/node/opentelemetry-instrumentation-restify/package.json#L66) which is a regular dependency. Your PR addresses the dev dependency on "restify" packages, which might be updated as well regardless IMO.

This seems reasonable to me, as version `4.3.4` is 4 years old and has only 20K downloads vs 313K for most popular version.

I wonder if the reason that `4.3.4` produced an error implies that the instrumentation no longer supports it's [advertised range of >= 4.0.0](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-restify#supported-versions)? (probably due to some other dependency that broke types?)

This instrumentation has no `test-all-versions` setup in place, so it's difficult to tell, but maybe you investigated this issue and can share insights?

**Comment by blumamir** [comment]
I think the original issue is in part due to the `@types/restify` package even being a regular dependency.
The only reason it's a dependency for end users (instead of dev dependency) is due to the template argument for InstrumentationBase [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/ec9ee131635dc2db88deea4f2efb887ff6f60577/plugins/node/opentelemetry-instrumentation-restify/src/instrumentation.ts#L37).

What we tend to do is to specify `any` type instead, and move the dependency to be dev-dependency, which can still break our development and CI, but not end users.

As this is related to #1132 , could be a good chance to fix this on the way if you are up for it. If not, just bumping the versions and verifing that v4 is still supported is also a great improvement.

**Comment by jscherer92** [comment]
I'm happy to do that fix. I know what I've done instead for this type of stuff is re export the type but if we just want to go with any that is fine.

Would we want unknown instead so to keep type safety?

File: plugins/node/opentelemetry-instrumentation-restify/package.json [hunk 2]
```diff
@@ -57,13 +57,13 @@
     "restify": "4.3.4",
     "rimraf": "3.0.2",
     "ts-mocha": "10.0.0",
-    "typescript": "4.3.5"
+    "typescript": "4.3.5",
+    "@types/restify": "4.3.8"
```
Comment (dyladan) [inline plugins/node/opentelemetry-instrumentation-restify/package.json:61]: If you move the types to dev dependencies I believe users have compilation failures. That's why we moved them to regular dependencies to begin with.

Comment (jscherer92) [reply plugins/node/opentelemetry-instrumentation-restify/package.json:61]: I'll take a look and see if that happens locally.

**Pull request closed without merging**